src/test/app/Credentials_test.cpp

```diff
@@ -0,0 +1,770 @@
+                env.close();
+
+                env(credentials::accept(subject, issuer, credType),
+                    ter(tecINSUFFICIENT_RESERVE));
+                env.close();
+            }
+        }
+
+        {
+            using namespace jtx;
+            Env env{*this, features};
+
+            env.fund(XRP(5000), subject, issuer);
+            env.close();
+
+            {
+                env(credentials::create(subject, issuer, credType));
+                env.close();
+
+                testcase("CredentialsAccept fail, invalid fee.");
+                auto jv = credentials::accept(subject, issuer, credType);
+                jv[jss::Fee] = -1;
+                env(jv, ter(temBAD_FEE));
+
+                testcase("CredentialsAccept fail, lsfAccepted already set.");
+                env(credentials::accept(subject, issuer, credType));
+                env.close();
+                env(credentials::accept(subject, issuer, credType),
+                    ter(tecDUPLICATE));
+                env.close();
+            }
+
+            {
+                const char credType2[] = "efghi";
+
+                testcase("CredentialsAccept fail, expired credentials.");
+                auto jv = credentials::create(subject, issuer, credType2);
+                uint32_t const t = env.now().time_since_epoch().count();
+                jv[sfExpiration.jsonName] = t;
+                env(jv);
+                env.close();
+
+                // credentials are expired now
+                env(credentials::accept(subject, issuer, credType2),
+                    ter(tecEXPIRED));
+                env.close();
+
+                // check that expired credentials were deleted
+                auto const jDelCred =
+                    credentials::ledgerEntry(env, subject, issuer, credType2);
+                BEAST_EXPECT(
+                    jDelCred.isObject() && jDelCred.isMember(jss::result) &&
+                    jDelCred[jss::result].isMember(jss::error) &&
+                    jDelCred[jss::result][jss::error] == "entryNotFound");
+            }
+        }
+
+        {
+            using namespace jtx;
+            Env env{*this, features};
+
+            env.fund(XRP(5000), issuer, subject, other);
+            env.close();
+
+            {
+                testcase("CredentialsAccept fail, issuer doesn't exist.");
+                auto jv = credentials::create(subject, issuer, credType);
+                env(jv);
+                env.close();
+
+                // delete issuer
+                int const delta = env.seq(issuer) + 255;
+                for (int i = 0; i < delta; ++i)
+                    env.close();
+                auto const acctDelFee{drops(env.current()->fees().increment)};
+                env(acctdelete(issuer, other), fee(acctDelFee));
+
+                // can't accept - no issuer account
+                jv = credentials::accept(subject, issuer, credType);
+                env(jv, ter(tecNO_ISSUER));
```

Comment:
Wouldn't the credential have been deleted anyways? So it should return this error even if the credential was never created?

Reply:
ye